satpy/readers/aapp_l1b.py

```diff
@@ -208,8 +208,12 @@ def _interpolate_arrays(self, *input_arrays):
         return output_arrays
 
     @staticmethod
-    def _create_40km_interpolator(lines, *arrays_40km):
-        from geotiepoints.interpolator import Interpolator
+    def _create_40km_interpolator(lines, *arrays_40km, geolocation=False):
+        if geolocation:
+            # Slower but accurate at datum line
+            from geotiepoints.geointerpolator import GeoInterpolator as Interpolator
+        else:
+            from geotiepoints.interpolator import Interpolator
```

Comment:
I don't think we need to keep the regular Interpolator as it seems to be unable to handle edge cases at all

Reply:
Not even for the angles?

Reply:
It will be the same problem with the angles actually, so it's probably best to use GeoInterpolator for the angles too (by adding an offset?)

Reply:
Just for reference: the aapp data only contains the azimuth difference, so GeoInterpolator cannot be used for the angles. Another solution for correct interpolation of the angles will be investigated in another PR.